internal/graph/graph.go

```diff
@@ -122,10 +149,37 @@ func BuildConnectedObjectGraph(typesystem *typesystem.TypeSystem) *ConnectedObje
 //  1. If parent's types includes `folder` type, and `folder` contains `viewer` relation then this is exactly a ttu rewrite and....?
 //  2. Otherwise, suppose the types contains `objectType` which has a relation `viewer`, then recurse on `objectType#viewer, folder#viewer`
 func (g *ConnectedObjectGraph) RelationshipIngresses(target *openfgapb.RelationReference, source *openfgapb.RelationReference) ([]*RelationshipIngress, error) {
```

Comment:
Technically, no. But all of the underlying code paths for it are.

Reply:
that would mean that we can remove the `resolveAllIngresses` const

Reply:
I'm all for removing unused code, believe me. However, in this case I think it is warranted to keep it, because we can use RelationshipIngresses for other potential optimizations in the future, whereas PrunedRelationshipIngresses wouldn't necessarily give all of the ingress targets possible, and so being a subset of the functionality is more limiting.

I think it's important for the code in the `graph` package to have support for both code paths (full ingress list via RelationshipIngresses AND a pruned ingress list via PrunedRelationshipIngresses). My fear is that if we remove the code then our implementation will become so specific to our current implementation behavior of ListObjects (with PrunedRelationshipIngresses), that we'll lose the functionality of the more general RelationshipIngresses. So this is why I feel it is warranted to keep it around in this case. Otherwise I'd say let's nuke it.

Reply:
Would it be doable to have just one `RelationshipIngresses` method and have it accept a parameter `findIngressOption`?

Reply:
It's a named helper function. What's wrong with that? It helps convey the difference in underlying behavior.

`PrunedRelationshipIngresses` and `RelationshipIngresses` both delegate to the underlying `findIngresses(..., findIngressOption)` function where `findIngressOption` is either `resolveAnyIngress` or `resolveAllIngresses`.

Reply:
It feels weird having an unused function lying around. If we 'merge' both uses into one function with an extra parameter, it won't be unused anymore

Reply:
I'm happy to change that, but do you mind if I save that for a quick follow up PR? Because that's going to require some rework on the test setup/structure etc..

If you feel strongly it should change in the scope of this PR (I understand why you may feel that way), then I'm happy to do it. But if it's not a big blocker for you, then it'd be nice to come back to that shortly after.

Reply:
Sounds good

Reply:
I'll create an issue to track this work once this is merged 👍